mmf/datasets/processors/frcnn_processor.py

```diff
@@ -0,0 +1,173 @@
+            .to(self.device)
+            .view(len(cfg.model.pixel_std), 1, 1)
+        )
+        self.normalizer = lambda x: (x - self.pixel_mean) / self.pixel_std
+
+    def pad(self, images: List[torch.Tensor]):
+        max_size = tuple(max(s) for s in zip(*[img.shape for img in images]))
+        image_sizes = [im.shape[-2:] for im in images]
+        images = [
+            F.pad(
+                im,
+                [0, max_size[-1] - size[1], 0, max_size[-2] - size[0]],
+                value=self.pad_value,
+            )
+            for size, im in zip(image_sizes, images)
+        ]
+
+        return torch.stack(images), torch.tensor(image_sizes)
+
+    def __call__(self, images: torch.Tensor, single_image: bool = False):
+        with torch.no_grad():
+            if not isinstance(images, list):
+                images = [images]
+            if single_image:
+                assert len(images) == 1
+            for i in range(len(images)):
+                if isinstance(images[i], torch.Tensor):
+                    images.insert(i, images.pop(i).to(self.device).float())
+                elif not isinstance(images[i], torch.Tensor):
+                    images.insert(
+                        i,
+                        torch.as_tensor(
+                            img_tensorize(images.pop(i), input_format=self.input_format)
+                        )
+                        .to(self.device)
+                        .float(),
+                    )
+            # resize smallest edge
+            raw_sizes = torch.tensor([im.shape[:2] for im in images])
+            images = self.aug(images)
+
+            # flip rgb to bgr
+            for idx in range(len(images)):
+                images[idx] = torch.flip(images[idx], [0])
+            # transpose images and convert to torch tensors
+            # images = [torch.as_tensor(i.astype("float32"))
+            # .permute(2, 0, 1).to(self.device) for i in images]
+            # now normalize before pad to avoid useless arithmetic
+            images = [self.normalizer(x) for x in images]
+            # now pad them to do the following operations
+            images, sizes = self.pad(images)
+            # Normalize
+
+            if self.size_divisibility > 0:
+                raise NotImplementedError()
+            # pad
+            scales_yx = torch.true_divide(raw_sizes, sizes)
+            if single_image:
+                return images[0], sizes[0], scales_yx[0]
+            else:
+                return images, sizes, scales_yx
+
+
+def get_image_from_url(url: str):
+    response = requests.get(url)
+    img = np.array(Image.open(BytesIO(response.content)))
+    return img
+
+
+def img_tensorize(im: str):
+    assert isinstance(im, str)
+    if os.path.isfile(im):
+        img = np.array(Image.open(im))
+    else:
+        img = get_image_from_url(im)
+        assert img is not None, f"could not connect to: {im}"
+    return img
```

Comment:
I have seen these two somewhere else as well. Please reuse or move to a image utils and then refactor and reuse.